core/direct-rpc-server/src/rpc_ws_handler.rs

```diff
@@ -69,21 +69,20 @@ where
 
 		debug!("RPC response string: {:?}", maybe_rpc_response);
 
-		let rpc_response = serde_json::from_str(
-			maybe_rpc_response.clone().unwrap_or_default().as_str(),
-		)
-		.map_err(|e| WebSocketError::Other(Box::new(DirectRpcError::SerializationError(e))))?;
-
-		match self.connection_watcher.must_be_watched(&rpc_response) {
-			Ok(maybe_connection_hash) =>
-				if let Some(connection_hash) = maybe_connection_hash {
-					self.connection_registry.store(
-						connection_hash,
-						connection_token.into(),
-						rpc_response,
-					);
-				},
-			Err(e) => return Err(WebSocketError::Other(Box::new(e))),
+		if let Ok(rpc_response) =
+			serde_json::from_str(maybe_rpc_response.clone().unwrap_or_default().as_str())
+		{
+			match self.connection_watcher.must_be_watched(&rpc_response) {
+				Ok(maybe_connection_hash) =>
+					if let Some(connection_hash) = maybe_connection_hash {
+						self.connection_registry.store(
+							connection_hash,
+							connection_token.into(),
+							rpc_response,
+						);
+					},
+				Err(e) => return Err(WebSocketError::Other(Box::new(e))),
+			}
```

Comment:
Yes I think this is much better - we only use the `RpcResponse` to determine if a call must be watched. So it shouldn't prevent us from sending a proper response 👍